src/classes/frDonation.cls

```diff
@@ -53,84 +54,136 @@ public class frDonation extends frModel implements frSyncable {
         if(!campaignMappingDisabled) {
             String campaignId = String.valueOf(request.get('campaignGoalId'));
             if (String.isNotBlank(campaignId)) {
                 List<Campaign> campaigns = [SELECT Id FROM Campaign WHERE fr_ID__c = :campaignId];
                 if (campaigns.size() > 0) {
-                    o.CampaignId = campaigns[0].Id;
+                    o.put('CampaignId', campaigns[0].Id);
                 } else {
                     frUtil.logRelationshipError(getFrType(), funraiseId, 
                                                 frUtil.Entity.CAMPAIGN, campaignId);
                 }
             }
         }
-        
+        //connect subscription
         String subscriptionId = String.valueOf(request.get('subscriptionId'));
         if (String.isNotBlank(subscriptionId)) {
-            List<Subscription__c> subscriptions = [SELECT Id FROM Subscription__c WHERE fr_ID__c = :subscriptionId];
+            List<SObject> subscriptions = Database.query('SELECT Id FROM ' + subsciptionObjName + ' WHERE fr_ID__c = :subscriptionId');
             if (subscriptions.size() > 0) {
-                o.Subscription__c = subscriptions.get(0).Id;
+                if(!frUtil.hasNPCobjects()) o.put('Subscription__c', subscriptions.get(0).get('Id'));
+                else o.put('GiftCommitmentId', subscriptions.get(0).get('Id'));
             } else {
                 frUtil.logRelationshipError(getFrType(), funraiseId, 
                                             frUtil.Entity.SUBSCRIPTION, subscriptionId);
             }
         }
-        Opportunity opp = this.o;
-        
-        
+
+        //connect pledge
         Boolean isPledge = Boolean.valueOf(request.get('pledge'));
-        Boolean alreadyMappedToPledge = [SELECT COUNT() FROM Opportunity 
-                                         WHERE fr_Id__c = :funraiseId AND Funraise_Pledge__c != null] > 0;
-        if(!isPledge && !alreadyMappedToPledge) {
-            Pledge__c pledge = frPledge.findActive(opp.fr_Donor__c, opp.CloseDate);
-            opp.Funraise_Pledge__c = pledge != null ? pledge.Id : null;
+        if(frUtil.hasNPCobjects()) {
+            List<SObject> alreadyMappedToPledgeList= Database.query('SELECT id FROM GiftTransaction WHERE fr_Id__c = :funraiseId AND Funraise_Pledge__c != null');
+            Boolean alreadyMappedToPledge = alreadyMappedToPledgeList.size() > 0;
+            if(!isPledge && !alreadyMappedToPledge) {
+                Pledge__c pledge = frPledge.findActiveAcc((Id) o.get('DonorId'), (Date) o.get('CheckDate'));
+                NPC_Setting__c npcSetting = NPC_Setting__c.getOrgDefaults();
+                if (npcSetting != null && pledge != null) {
+                    String pledgeId = npcSetting.NPCFieldName__c;
+                    o.put('Funraise_GT_Pledge__c', pledge.id);
+                } else {
+                    System.debug('NPC Setting not found at the organization level.');
```

Comment:
I think we should remove this System.debug and replace it with some logging using the funraise_Error__c object if there is something we need the user to do

Reply:
Changed to use the frUtil.logRelationshipError method!